# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.cpp
# Commit: d5b7fe789001
# Full Hash: d5b7fe789001225baaab1ceee4ed1f8ec9021292
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Mark mCapturedFrameInvalidated even if mIsCanvasDirty. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79014
# ==============================================================================

diff -r 64fbb616a0f3 -r d5b7fe789001 dom/canvas/WebGLContext.cpp
--- a/dom/canvas/WebGLContext.cpp	Thu Jun 11 06:37:43 2020 +0000
+++ b/dom/canvas/WebGLContext.cpp	Thu Jun 11 06:37:50 2020 +0000
@@ -248,13 +248,13 @@
 }
 
 void ClientWebGLContext::MarkCanvasDirty() {
-  if (mIsCanvasDirty) return;
-  mIsCanvasDirty = true;
-
   if (!mCanvasElement) return;
 
   mCapturedFrameInvalidated = true;
 
+  if (mIsCanvasDirty) return;
+  mIsCanvasDirty = true;
+
   SVGObserverUtils::InvalidateDirectRenderingObservers(mCanvasElement);
 
   mCanvasElement->InvalidateCanvasContent(nullptr);
